p2psrv/server.go

```diff
@@ -37,8 +37,26 @@ type Server struct {
 	dialingNodes    *nodeMap
 }
 
-// New create a p2p server.
+// New creates a server with a default p2p.Server.
 func New(opts *Options) *Server {
+	return NewWithServer(opts, &p2p.Server{
+		Config: p2p.Config{
+			Name:        opts.Name,
+			PrivateKey:  opts.PrivateKey,
+			MaxPeers:    opts.MaxPeers,
+			NoDiscovery: true,  // disable discovery inside p2p.Server instance(we use our own)
+			DiscoveryV5: false, // disable discovery inside p2p.Server instance(we use our own)
+			ListenAddr:  opts.ListenAddr,
+			NetRestrict: opts.NetRestrict,
+			NAT:         opts.NAT,
+			NoDial:      opts.NoDial,
+			DialRatio:   int(math.Sqrt(float64(opts.MaxPeers))),
+		},
+	})
+}
+
+// NewWithServer creates a server with a provided p2p.Server.
+func NewWithServer(opts *Options, srv *p2p.Server) *Server {
```

Comment:
Could you elaborate use cases on this? Looks like the `p2p.Server` are created based on the `opts`.